docs/architecture/porting-existing-aspnet-apps/more-migration-scenarios.md

```diff
@@ -0,0 +1,122 @@
+---
+title: More migration scenarios
+description: This section describes additional migration scenarios and techniques for upgrading .NET Framework apps to .NET Core / .NET 5.
+author: ardalis
+ms.date: 02/11/2021
+---
+
+# More migration scenarios
+
+[!INCLUDE [book-preview](../../../includes/book-preview.md)]
+
+This section describes several different ASP.NET app scenarios and situation, and offers specific techniques for solving each of them. You can use this section to identify scenarios applicable to your app and evaluate which techniques will work for your app and its hosting environment.
+
+## Migrate ASP.NET MVC 5 and WebApi 2 to ASP.NET Core MVC
+
+A common scenario in ASP.NET MVC 5 and Web API 2 apps was for both products to be installed in the same application. This is a supported and relatively common approach used by many teams, but because the two products use different abstractions, there is some redundant effort needed. For example, setting up routes for ASP.NET MVC is done using methods on `RouteCollection`, such as `MapMvcAttributeRoutes()` and `MapRoute()`. But ASP.NET Web API 2 routing is managed with `HttpConfiguration` and methods like `MapHttpAttributeRoutes()` and `MapHttpRoute()`.
+
+The `eShopLegacyMVC` app includes both ASP.NET MVC and Web API, and includes methods in its `App_Start` folder for setting up routes for both. It also supports dependency injection using Autofac, which also requires two sets of similar work to configure:
+
+```csharp
+protected IContainer RegisterContainer()
+{
+    var builder = new ContainerBuilder();
+
+    var thisAssembly = Assembly.GetExecutingAssembly();
+    builder.RegisterControllers(thisAssembly);      // MVC controllers
+    builder.RegisterApiControllers(thisAssembly);   // Web API controllers
+
+    var mockData = bool.Parse(ConfigurationManager.AppSettings["UseMockData"]);
+    builder.RegisterModule(new ApplicationModule(mockData));
+
+    var container = builder.Build();
+
+    // set mvc resolver
+    DependencyResolver.SetResolver(new AutofacDependencyResolver(container));
+
+    // set webapi resolver
+    var resolver = new AutofacWebApiDependencyResolver(container);
+    GlobalConfiguration.Configuration.DependencyResolver = resolver;
+
+    return container;
+}
+```
+
+When upgrading these apps to use ASP.NET Core, this duplicate effort and the confusion that sometimes accompanies it is eliminated. ASP.NET Core MVC is a unified framework with one set of rules for routing, filters, and more. Dependency injection is built into ASP.NET Core itself. All of this can can be configured in `Startup.cs`, as is shown in the `eShopPorted` app in the sample.
```

Comment:
```suggestion
When upgrading these apps to use ASP.NET Core, this duplicate effort and the confusion that sometimes accompanies it is eliminated. ASP.NET Core MVC is a unified framework with one set of rules for routing, filters, and more. Dependency injection is built into .NET Core itself. All of this can can be configured in `Startup.cs`, as is shown in the `eShopPorted` app in the sample.
```